### ISSUE TITLE
Issue running on Docker\Homeassistant\PyHap

### ISSUE THREAD

**Issue by stich86** [jstasiak/python-zeroconf#157]
Hi all,

i'm going crazy how to solve the isssue that i think it's related to this library. Essentialy the HA is using the PyHap library (https://github.com/ikalchev/HAP-python/issues/176) to publish mDNS records and using HomeKit on iPhone. As i've debugged it seems that the zeroconf part publish correct mDNS entry and after some time (in my case few minutes) remove the entry. This cause my Home App to put all accessory on "No Response".

I'm currently use Docker on Synology NAS with "host" network, so there is no nat or other overlay.

Checking with "avahi-cache" i see this behaviour:

```
+ ovs_eth0 IPv4 Home Assistant Bridge                         _hap._tcp            local
+ ovs_eth0-2 IPv4 Home Assistant Bridge                         _hap._tcp            local
- ovs_eth0-2 IPv4 Home Assistant Bridge                         _hap._tcp            local
- ovs_eth0 IPv4 Home Assistant Bridge                         _hap._tcp            local
+ ovs_eth0 IPv4 Home Assistant Bridge                         _hap._tcp            local
+ ovs_eth0-2 IPv4 Home Assistant Bridge                         _hap._tcp            local
- ovs_eth0-2 IPv4 Home Assistant Bridge                         _hap._tcp            local
- ovs_eth0 IPv4 Home Assistant Bridge                         _hap._tcp            local
```

Can someone help me to check how can cause this issue? I've also tried to rollback to various version of this library but without success :(

Thanks!

**Comment by jstasiak** [comment]
Can you provide the full timestamped zeroconf debug log? (produced by the application using this library, that is)

https://github.com/jstasiak/python-zeroconf/blob/4218d757994467ee710b0cad034ea1fb6035d3ea/examples/browser.py#L40 is how you  set the zeroconf logger to the debug level

Maybe there's something obvious in there.

**Comment by stich86** [comment]
ok finally i've found to put in debug

`
2019-03-11 15:35:15 DEBUG (zeroconf-Engine) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[srv,in-unique,Home Assistant Bridge._hap._tcp.local.,120/119.99893872070312,lynx-DS.local.:51827], 0), (record[txt,in-unique,Home Assistant Bridge._hap._tcp.local.,120/119.9989560546875,b'\x18md=Hom']..., 0)], authorities=[], additionals=[10.0.1.100]}> (195 bytes) as b'\x00\x00\x84\x00\x00\x00\x00\x02\x00\x00\x00\x01\x15Home Assistant Bridge\x04_hap\x04_tcp\x05local\x00\x00!\x80\x01\x00\x00\x00x\x00\x10\x00\x00\x00\x00\xcas\x07lynx-DS\xc0,\xc0\x0c\x00\x10\x80\x01\x00\x00\x00x\x00Z\x18md=Home Assistant Bridge\x06pv=1.0\x14id=23:F2:B3:0A:31:78\x04c#=2\x04s#=1\x04ff=0\x04ci=2\x04sf=0\x0bsh=1SnO8g==\xc0C\x00\x01\x80\x01\x00\x00\x00x\x00\x04\n\x00\x01d'...
2019-03-11 15:35:15 DEBUG (zeroconf-Engine) [zeroconf] Received from '127.0.0.1':5353: b'\x00\x00\x84\x00\x00\x00\x00\x02\x00\x00\x00\x01\x15Home Assistant Bridge\x04_hap\x04_tcp\x05local\x00\x00!\x80\x01\x00\x00\x00x\x00\x10\x00\x00\x00\x00\xcas\x07lynx-DS\xc0,\xc0\x0c\x00\x10\x80\x01\x00\x00\x00x\x00Z\x18md=Home Assistant Bridge\x06pv=1.0\x14id=23:F2:B3:0A:31:78\x04c#=2\x04s#=1\x04ff=0\x04ci=2\x04sf=0\x0bsh=1SnO8g==\xc0C\x00\x01\x80\x01\x00\x00\x00x\x00\x04\n\x00\x01d'
2019-03-11 15:35:15 DEBUG (zeroconf-Engine) [zeroconf] Received from '10.0.1.100':5353: b'\x00\x00\x84\x00\x00\x00\x00\x02\x00\x00\x00\x01\x15Home Assistant Bridge\x04_hap\x04_tcp\x05local\x00\x00!\x80\x01\x00\x00\x00x\x00\x10\x00\x00\x00\x00\xcas\x07lynx-DS\xc0,\xc0\x0c\x00\x10\x80\x01\x00\x00\x00x\x00Z\x18md=Home Assistant Bridge\x06pv=1.0\x14id=23:F2:B3:0A:31:78\x04c#=2\x04s#=1\x04ff=0\x04ci=2\x04sf=0\x0bsh=1SnO8g==\xc0C\x00\x01\x80\x01\x00\x00\x00x\x00\x04\n\x00\x01d'
2019-03-11 15:35:15 DEBUG (zeroconf-Engine) [zeroconf] Received from '172.17.0.1':5353: b'\x00\x00\x84\x00\x00\x00\x00\x02\x00\x00\x00\x01\x15Home Assistant Bridge\x04_hap\x04_tcp\x05local\x00\x00!\x80\x01\x00\x00\x00x\x00\x10\x00\x00\x00\x00\xcas\x07lynx-DS\xc0,\xc0\x0c\x00\x10\x80\x01\x00\x00\x00x\x00Z\x18md=Home Assistant Bridge\x06pv=1.0\x14id=23:F2:B3:0A:31:78\x04c#=2\x04s#=1\x04ff=0\x04ci=2\x04sf=0\x0bsh=1SnO8g==\xc0C\x00\x01\x80\x01\x00\x00\x00x\x00\x04\n\x00\x01d'
2019-03-11 15:35:16 DEBUG (Thread-8) [pyhap.hap_server] Pair verify [1/2].
2019-03-11 15:35:16 DEBUG (Thread-8) [pyhap.hap_server] Pair verify [2/2]
2019-03-11 15:35:16 DEBUG (Thread-8) [pyhap.accessory_driver] Get accessories response: [{'aid': 1, 'services': [{'iid': 1, 'type': '0000003E-0000-1000-8000-0026BB765291', 'characteristics': [{'iid': 2, 'type': '00000014-0000-1000-8000-0026BB765291', 'description': 'Identify', 'perms': ['pw'], 'format': 'bool'}, {'iid': 3, 'type': '00000020-0000-1000-8000-0026BB765291', 'description': 'Manufacturer', 'perms': ['pr'], 'format': 'string', 'value': 'Home Assistant'}, {'iid': 4, 'type': '00000021-0000-1000-8000-0026BB765291', 'description': 'Model', 'perms': ['pr'], 'format': 'string', 'value': 'Bridge'}, {'iid': 5, 'type': '00000023-0000-1000-8000-0026BB765291', 'description': 'Name', 'perms': ['pr'], 'format': 'string', 'value': 'Home Assistant Bridge'}, {'iid': 6, 'type': '00000030-0000-1000-8000-0026BB765291', 'description': 'SerialNumber', 'perms': ['pr'], 'format': 'string', 'value': 'homekit.bridge'}, {'iid': 7, 'type': '00000052-0000-1000-8000-0026BB765291', 'description': 'FirmwareRevision', 'perms': ['pr'], 'format': 'string', 'value': '0.89.1'}]}]}]
2019-03-11 15:35:26 DEBUG (zeroconf-Engine) [zeroconf] Received from '10.0.1.100':5353: b'\x00\x00\x00\x00\x00\x01\x00\x0e\x00\x00\x00\x00\t_services\x07_dns-sd\x04_udp\x05local\x00\x00\x0c\x00\x01\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\x10\x08_homekit\x04_tcp\xc0#\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\x08\x05_http\xc0C\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\t\x06_adisk\xc0C\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\x0f\x0c_device-info\xc0C\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\x0e\x0b_afpovertcp\xc0C\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\x07\x04_smb\xc0C\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\x0f\x0c_workstation\xc0C\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\x0e\x0b_touch-able\xc0C\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\x0e\x0b_appletv-v2\xc0C\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\x0f\x0c_sleep-proxy\xc0\x1e\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\x12\x0f_companion-link\xc0C\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\x11\x0e_mediaremotetv\xc0C\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\x0b\x08_airplay\xc0C\xc0\x0c\x00\x0c\x00\x01\x00\x00\x11\x94\x00\x08\x05_raop\xc0C'
2019-03-11 15:35:26 DEBUG (zeroconf-Engine) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[ptr,in,_services._dns-sd._udp.local.,120/119.99845239257813,_hap._tcp.local.], 0)], authorities=[], additionals=[]}> (64 bytes) as b'\x00\x00\x84\x00\x00\x00\x00\x01\x00\x00\x00\x00\t_services\x07_dns-sd\x04_udp\x05local\x00\x00\x0c\x00\x01\x00\x00\x00x\x00\x0c\x04_hap\x04_tcp\xc0#'...
`

so in the first attempt zeroconf publish the right record, but after few second it create a new record that delete the old one. It's a loop.

Do you think it's a problem of the library? Let me know if you need other logs

thanks!

**Comment by stich86** [comment]
so i've done a little bit debug checking better the logs:

when start-up HA on first attempt ((and HK works for few seconds) this is the log:

```
2019-03-11 15:45:53 DEBUG (SyncWorker_0) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=1024, questions=[question[ptr,in,_hap._tcp.local.]], answers=[], authorities=[record[ptr,in,_hap._tcp.local.,120/119.99849389648438,Home Assistant Bridge._hap._tcp.local.]], additionals=[]}> (69 bytes) as b'\x00\x00\x04\x00\x00\x01\x00\x00\x00\x01\x00\x00\x04_hap\x04_tcp\x05local\x00\x00\x0c\x00\x01\xc0\x0c\x00\x0c\x00\x01\x00\x00\x00x\x00\x18\x15Home Assistant Bridge\xc0\x0c'...
2019-03-11 15:45:53 DEBUG (SyncWorker_0) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=1024, questions=[question[ptr,in,_hap._tcp.local.]], answers=[], authorities=[record[ptr,in,_hap._tcp.local.,120/119.9979853515625,Home Assistant Bridge._hap._tcp.local.]], additionals=[]}> (69 bytes) as b'\x00\x00\x04\x00\x00\x01\x00\x00\x00\x01\x00\x00\x04_hap\x04_tcp\x05local\x00\x00\x0c\x00\x01\xc0\x0c\x00\x0c\x00\x01\x00\x00\x00x\x00\x18\x15Home Assistant Bridge\xc0\x0c'...
2019-03-11 15:45:54 DEBUG (SyncWorker_0) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=1024, questions=[question[ptr,in,_hap._tcp.local.]], answers=[], authorities=[record[ptr,in,_hap._tcp.local.,120/119.99870947265624,Home Assistant Bridge._hap._tcp.local.]], additionals=[]}> (69 bytes) as b'\x00\x00\x04\x00\x00\x01\x00\x00\x00\x01\x00\x00\x04_hap\x04_tcp\x05local\x00\x00\x0c\x00\x01\xc0\x0c\x00\x0c\x00\x01\x00\x00\x00x\x00\x18\x15Home Assistant Bridge\xc0\x0c'...
2019-03-11 15:45:54 DEBUG (SyncWorker_0) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[ptr,in,_hap._tcp.local.,120/119.9968837890625,Home Assistant Bridge._hap._tcp.local.], 0), (record[srv,in,Home Assistant Bridge._hap._tcp.local.,120/119.99685620117188,lynx-DS.local.:51827], 0), (record[txt,in,Home Assistant Bridge._hap._tcp.local.,120/119.99685498046875,b'\x18md=Hom']..., 0), (10.0.1.100, 0)], authorities=[], additionals=[]}> (209 bytes) as b"\x00\x00\x84\x00\x00\x00\x00\x04\x00\x00\x00\x00\x04_hap\x04_tcp\x05local\x00\x00\x0c\x00\x01\x00\x00\x00x\x00\x18\x15Home Assistant Bridge\xc0\x0c\xc0'\x00!\x00\x01\x00\x00\x00x\x00\x10\x00\x00\x00\x00\xcas\x07lynx-DS\xc0\x16\xc0'\x00\x10\x00\x01\x00\x00\x00x\x00Z\x18md=Home Assistant Bridge\x06pv=1.0\x14id=23:F2:B3:0A:31:78\x04c#=2\x04s#=1\x04ff=0\x04ci=2\x04sf=0\x0bsh=z4Qa1g==\xc0Q\x00\x01\x00\x01\x00\x00\x00x\x00\x04\n\x00\x01d"...
2019-03-11 15:45:54 DEBUG (SyncWorker_0) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[ptr,in,_hap._tcp.local.,120/119.99868969726562,Home Assistant Bridge._hap._tcp.local.], 0), (record[srv,in,Home Assistant Bridge._hap._tcp.local.,120/119.99867065429687,lynx-DS.local.:51827], 0), (record[txt,in,Home Assistant Bridge._hap._tcp.local.,120/119.99865991210937,b'\x18md=Hom']..., 0), (10.0.1.100, 0)], authorities=[], additionals=[]}> (209 bytes) as b"\x00\x00\x84\x00\x00\x00\x00\x04\x00\x00\x00\x00\x04_hap\x04_tcp\x05local\x00\x00\x0c\x00\x01\x00\x00\x00x\x00\x18\x15Home Assistant Bridge\xc0\x0c\xc0'\x00!\x00\x01\x00\x00\x00x\x00\x10\x00\x00\x00\x00\xcas\x07lynx-DS\xc0\x16\xc0'\x00\x10\x00\x01\x00\x00\x00x\x00Z\x18md=Home Assistant Bridge\x06pv=1.0\x14id=23:F2:B3:0A:31:78\x04c#=2\x04s#=1\x04ff=0\x04ci=2\x04sf=0\x0bsh=z4Qa1g==\xc0Q\x00\x01\x00\x01\x00\x00\x00x\x00\x04\n\x00\x01d"...
2019-03-11 15:45:54 DEBUG (SyncWorker_0) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[ptr,in,_hap._tcp.local.,120/119.99868603515625,Home Assistant Bridge._hap._tcp.local.], 0), (record[srv,in,Home Assistant Bridge._hap._tcp.local.,120/119.99865112304687,lynx-DS.local.:51827], 0), (record[txt,in,Home Assistant Bridge._hap._tcp.local.,120/119.9986396484375,b'\x18md=Hom']..., 0), (10.0.1.100, 0)], authorities=[], additionals=[]}> (209 bytes) as b"\x00\x00\x84\x00\x00\x00\x00\x04\x00\x00\x00\x00\x04_hap\x04_tcp\x05local\x00\x00\x0c\x00\x01\x00\x00\x00x\x00\x18\x15Home Assistant Bridge\xc0\x0c\xc0'\x00!\x00\x01\x00\x00\x00x\x00\x10\x00\x00\x00\x00\xcas\x07lynx-DS\xc0\x16\xc0'\x00\x10\x00\x01\x00\x00\x00x\x00Z\x18md=Home Assistant Bridge\x06pv=1.0\x14id=23:F2:B3:0A:31:78\x04c#=2\x04s#=1\x04ff=0\x04ci=2\x04sf=0\x0bsh=z4Qa1g==\xc0Q\x00\x01\x00\x01\x00\x00\x00x\x00\x04\n\x00\x01d"...
```

Still don't know why it does the query three times

After few minutes it starting send wrong records:

> 2019-03-11 15:49:42 DEBUG (zeroconf-Engine) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[ptr,in,_services._dns-sd._udp.local.,120/119.9989384765625,_hap._tcp.local.], 0)], authorities=[], additionals=[]}> (64 bytes) as b'\x00\x00\x84\x00\x00\x00\x00\x01\x00\x00\x00\x00\t_services\x07_dns-sd\x04_udp\x05local\x00\x00\x0c\x00\x01\x00\x00\x00x\x00\x0c\x04_hap\x04_tcp\xc0#'...
> 2019-03-11 15:49:42 DEBUG (zeroconf-Engine) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[ptr,in,_hap._tcp.local.,120/119.99921264648438,Home Assistant Bridge._hap._tcp.local.], 0)], authorities=[], additionals=[]}> (63 bytes) as b'\x00\x00\x84\x00\x00\x00\x00\x01\x00\x00\x00\x00\x04_hap\x04_tcp\x05local\x00\x00\x0c\x00\x01\x00\x00\x00x\x00\x18\x15Home Assistant Bridge\xc0\x0c'...
> 2019-03-11 15:49:42 DEBUG (zeroconf-Engine) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[srv,in-unique,Home Assistant Bridge._hap._tcp.local.,120/119.99889794921874,lynx-DS.local.:51827], 0), (record[txt,in-unique,Home Assistant Bridge._hap._tcp.local.,120/119.99891528320312,b'\x18md=Hom']..., 0)], authorities=[], additionals=[10.0.1.100]}> (195 bytes) as b'\x00\x00\x84\x00\x00\x00\x00\x02\x00\x00\x00\x01\x15Home Assistant Bridge\x04_hap\x04_tcp\x05local\x00\x00!\x80\x01\x00\x00\x00x\x00\x10\x00\x00\x00\x00\xcas\x07lynx-DS\xc0,\xc0\x0c\x00\x10\x80\x01\x00\x00\x00x\x00Z\x18md=Home Assistant Bridge\x06pv=1.0\x14id=23:F2:B3:0A:31:78\x04c#=2\x04s#=1\x04ff=0\x04ci=2\x04sf=0\x0bsh=z4Qa1g==\xc0C\x00\x01\x80\x01\x00\x00\x00x\x00\x04\n\x00\x01d'...
> 2019-03-11 15:49:57 DEBUG (zeroconf-Engine) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[ptr,in,_hap._tcp.local.,120/119.99908959960938,Home Assistant Bridge._hap._tcp.local.], 0)], authorities=[], additionals=[]}> (63 bytes) as b'\x00\x00\x84\x00\x00\x00\x00\x01\x00\x00\x00\x00\x04_hap\x04_tcp\x05local\x00\x00\x0c\x00\x01\x00\x00\x00x\x00\x18\x15Home Assistant Bridge\xc0\x0c'...
> 2019-03-11 15:50:13 DEBUG (zeroconf-Engine) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[ptr,in,_hap._tcp.local.,120/119.9985830078125,Home Assistant Bridge._hap._tcp.local.], 0)], authorities=[], additionals=[]}> (63 bytes) as b'\x00\x00\x84\x00\x00\x00\x00\x01\x00\x00\x00\x00\x04_hap\x04_tcp\x05local\x00\x00\x0c\x00\x01\x00\x00\x00x\x00\x18\x15Home Assistant Bridge\xc0\x0c'...
> 2019-03-11 15:50:45 DEBUG (zeroconf-Engine) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[ptr,in,_services._dns-sd._udp.local.,120/119.99896166992187,_hap._tcp.local.], 0)], authorities=[], additionals=[]}> (64 bytes) as b'\x00\x00\x84\x00\x00\x00\x00\x01\x00\x00\x00\x00\t_services\x07_dns-sd\x04_udp\x05local\x00\x00\x0c\x00\x01\x00\x00\x00x\x00\x0c\x04_hap\x04_tcp\xc0#'...
> 2019-03-11 15:50:45 DEBUG (zeroconf-Engine) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[ptr,in,_hap._tcp.local.,120/119.99900390625,Home Assistant Bridge._hap._tcp.local.], 0)], authorities=[], additionals=[]}> (63 bytes) as b'\x00\x00\x84\x00\x00\x00\x00\x01\x00\x00\x00\x00\x04_hap\x04_tcp\x05local\x00\x00\x0c\x00\x01\x00\x00\x00x\x00\x18\x15Home Assistant Bridge\xc0\x0c'...
> 2019-03-11 15:51:49 DEBUG (zeroconf-Engine) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[ptr,in,_services._dns-sd._udp.local.,120/119.998896484375,_hap._tcp.local.], 0)], authorities=[], additionals=[]}> (64 bytes) as b'\x00\x00\x84\x00\x00\x00\x00\x01\x00\x00\x00\x00\t_services\x07_dns-sd\x04_udp\x05local\x00\x00\x0c\x00\x01\x00\x00\x00x\x00\x0c\x04_hap\x04_tcp\xc0#'...
> 2019-03-11 15:51:49 DEBUG (zeroconf-Engine) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[ptr,in,_hap._tcp.local.,120/119.99918969726562,Home Assistant Bridge._hap._tcp.local.], 0)], authorities=[], additionals=[]}> (63 bytes) as b'\x00\x00\x84\x00\x00\x00\x00\x01\x00\x00\x00\x00\x04_hap\x04_tcp\x05local\x00\x00\x0c\x00\x01\x00\x00\x00x\x00\x18\x15Home Assistant Bridge\xc0\x0c'...
> 2019-03-11 15:51:49 DEBUG (zeroconf-Engine) [zeroconf] Sending <DNSOutgoing:{multicast=True, flags=33792, questions=[], answers=[(record[srv,in-unique,Home Assistant Bridge._hap._tcp.local.,120/119.99901000976563,lynx-DS.local.:51827], 0), (record[txt,in-unique,Home Assistant Bridge._hap._tcp.local.,120/119.99902075195313,b'\x18md=Hom']..., 0)], authorities=[], additionals=[10.0.1.100]}> (195 bytes) as b'\x00\x00\x84\x00\x00\x00\x00\x02\x00\x00\x00\x01\x15Home Assistant Bridge\x04_hap\x04_tcp\x05local\x00\x00!\x80\x01\x00\x00\x00x\x00\x10\x00\x00\x00\x00\xcas\x07lynx-DS\xc0,\xc0\x0c\x00\x10\x80\x01\x00\x00\x00x\x00Z\x18md=Home Assistant Bridge\x06pv=1.0\x14id=23:F2:B3:0A:31:78\x04c#=2\x04s#=1\x04ff=0\x04ci=2\x04sf=0\x0bsh=z4Qa1g==\xc0C\x00\x01\x80\x01\x00\x00\x00x\x00\x04\n\x00\x01d'...

**Comment by stich86** [comment]
Bingo! I’ve and USG (unifi gateway) with mDNS reflector enabled to be able to communicate between VLAN. Disabling it and restarting the HA instance solve temporary the issue. It seems like that the library goes into a loop back when receive an outgoing answer back as response (because USG clone the mDNS request on all interfaces). Do you think it’s possible to implement a check that avoid this behavior? This problem seems not present on libavahi.

Thanks for your support!

**Comment by jstasiak** [comment]
Very nice debugging work! Yeah, I think it's doable, a pull request fixing this would definitely be welcome.

**Comment by bdraco** [comment]
This should be solved with 0.32.0+ as it implements duplicate question suppression and a variety of other traffic reduction techniques

**Comment by bdraco** [comment]
@stich86 let me know if this is still an issue. I'm going to close this now because I'm assuming it's solved as I've had a code to specifically address this case￼
